source/slang/parser.cpp

```diff
@@ -8,6 +8,52 @@
 
 namespace Slang
 {
+    // Pre-declare
+    static Name* getName(Parser* parser, String const& text);
+
+    // Helper class useful to build a list of modifiers. 
+    struct ModifierListBuilder
```

Comment:
Is the `ModifierListBuilder` type still needed? I'd never had cause to use it before, but I'm fine with having it here if it makes sense.

Reply:
I think it's a useful construct, it makes constructing a ModifierList simpler/cleaner. It can also be used for handling the SharedModifiers stuff, without searching (another version did do this). So I think a case can be made for it being used elsewhere.